library/core/src/main/java/com/google/android/exoplayer2/text/dvb/DvbParser.java

```diff
@@ -0,0 +1,1569 @@
+                reserved                            4
+                object_vertical_position           12     Specifies the vertical position of the top left pixel of this object
+                if (object_type ==0x01 or object_type == 0x02){ UNSUPPORTED
+                    foreground_pixel_code           8
+                    background_pixel_code           8
+                }
+            }
+        }
+        */
+
+    RegionComposition region = new RegionComposition();
+
+    region.pageId = tsStream.readBits(16);
+    int remainingSegmentLength = tsStream.readBits(16);
+    region.regionId = tsStream.readBits(8);
+    region.regionVersionNumber = tsStream.readBits(4);
+    if (tsStream.readBits(1) == 1) {
+      region.flags |= REGION_FILL_FLAG;
+    }
+    tsStream.skipBits(3);
+    region.regionWidth = tsStream.readBits(16);
+    region.regionHeight = tsStream.readBits(16);
+    region.regionLevelOfCompatibility = tsStream.readBits(3);
+    region.regionDepth = tsStream.readBits(3);
+    tsStream.skipBits(2);
+    region.clutId = tsStream.readBits(8);
+    tsStream.skipBits(16);
+
+    if (BuildConfig.DEBUG) {
+      Log.d(TAG, "        New Region Composition. regionId: " + region.regionId +
+          " (w/h): (" + region.regionWidth + "/" + region.regionHeight + ")");
+    }
+
+    int arrayIndex = 0; // index by an incremental counter to allow repeating objects in one region
+
+    if (subtitleService.pageComposition != null && subtitleService.pageComposition.pageId == region.pageId &&
+        subtitleService.pageComposition.pageState == DVBSUB_PCS_STATE_NORMAL) {
+      RegionComposition tempRegion = subtitleService.regions.get(region.regionId);
+      if (tempRegion != null) {
+        region.regionObjects = tempRegion.regionObjects;
+        arrayIndex = region.regionObjects.size();
+      }
+    }
+
+    remainingSegmentLength -= 10;
+    RegionObject object;
+    while (remainingSegmentLength > 0) {
+      object = new RegionObject();
+
+      object.objectId = tsStream.readBits(16);
+      object.objectType = tsStream.readBits(2);
+      object.objectProvider = tsStream.readBits(2);
+      object.objectHorizontalPosition = tsStream.readBits(12);
+      tsStream.skipBits(4);
+      object.objectVerticalPosition = tsStream.readBits(12);
+      remainingSegmentLength -= 6;
+
+      if (object.objectType == 0x01 || object.objectType == 0x02) { // Only seems to affect to char subtitles
+        object.foregroundPixelCode = tsStream.readBits(8);
+        object.backgroundPixelCode = tsStream.readBits(8);
+        remainingSegmentLength -= 2;
+      }
+
+      if (BuildConfig.DEBUG) {
+        Log.d(TAG, "            New Region Object[" + arrayIndex + "]." +
+            " objectId: " + object.objectId +
+            " (x/y): (" + object.objectHorizontalPosition + "/" + object.objectVerticalPosition + ")");
+      }
+
+      region.regionObjects.put(arrayIndex++, object);
```

Comment:
My question is whether it's intentional that both the local `object` and `subtitleService.regions.get(region.regionId)` are being modified on this line (because they're both referencing the exact same `regionObjects` array).  Intuitively I'd expect L783 to copy the elements from `tempRegion.regionObjects` into `region.regionObjects`, not just update the reference. That would result in only `object` being modified on this line.

Oh, right. Looking deeper it looks like `subtitleService.regions` is updated to remove `tempRegion` anyway after the call to this method complete at L523, so it doesn't make much difference.